ddtrace/opentracer/tracer.py

```diff
@@ -1,13 +1,68 @@
 import opentracing
+import logging
+
+from ddtrace import Tracer as DatadogTracer
+from ddtrace.constants import FILTERS_KEY
+from ddtrace.ext import AppTypes
+from ddtrace.settings import ConfigException
+
+from .constants import ConfigKeys as keys
+from .settings import config_invalid_keys
+
+from .util import merge_dicts
+
+log = logging.getLogger(__name__)
+
+DEFAULT_CONFIG = {
+    keys.AGENT_HOSTNAME: 'localhost',
+    keys.AGENT_PORT: 8126,
+    keys.DEBUG: False,
+    keys.ENABLED: True,
+    keys.GLOBAL_TAGS: {},
+    keys.SERVICE_NAME: None,
+    keys.SAMPLER: None,
+    keys.APP_TYPE: AppTypes.worker,
+    keys.CONTEXT_PROVIDER: None,
+    keys.PRIORITY_SAMPLING: None,
+    keys.SETTINGS: {
+        FILTERS_KEY: [],
+    },
+}
 
 
 class Tracer(opentracing.Tracer):
-    """"""
+    """A wrapper providing an OpenTracing API for the Datadog tracer."""
 
-    __slots__ = []
+    __slots__ = ['_enabled', '_debug', '_service_name', '_tracer']
 
-    def __init__(self, scope_manager=None):
-        pass
+    def __init__(self, config={}, service_name=None, scope_manager=None):
+        # Merge the given config with the default into a new dict
+        self._config = merge_dicts(DEFAULT_CONFIG, config)
+
+        # Pull out commonly used properties for performance
+        self._service_name = self._config.get(keys.SERVICE_NAME, None) or service_name
+        self._enabled = self._config.get(keys.ENABLED)
+        self._debug = self._config.get(keys.DEBUG)
+
+        if self._debug:
+            # ensure there are no typos in any of the keys
+            invalid_keys = config_invalid_keys(self._config)
+            if invalid_keys:
+                raise ConfigException('invalid keys given (%s)' % ','.join(invalid_keys))
```

Comment:
For code that supports both Python 2.x and 3.x you'll generally want to use [`.format`](https://docs.python.org/3.1/library/string.html#format-examples) instead of `%`

Reply:
👍